nixos/tests/dsl.nix

```diff
@@ -18,12 +18,12 @@ let
         default = name;
       };
       options.request = lib.mkEnableOption "Whether to mark the node for building";
-      options.needed = lib.mkOption {
+      options.assertNeeded = lib.mkOption {
```

Comment:
I just realized this isn't well defined: "needed" corresponds either to "will be built" or "will be fetched" in our test, and the later commit indeed ignores of the variants...